### PR TITLE
Add prop to make BEM optional

### PR DESCRIPTION
**Pull request by fpapado**
Hi folks,

We have a use case where we use atomic classes to define our styles. In the rest of our codebase, we have a granular way of overwriting props. When we need to use className as a string (such as when interfacing with react-tabs), we want to completely replace the base class, in order to overwrite styles properly.

For example, consider this (using `<div>` to simplify):
```js
// Common: display: inline-block and padding: 3; "dib pa3"

// idle
<div className="dib pa3 color-white"/>

// selected
<div className="dib pa3 color-blue"/>

// disabled
<div className="dib pa3 color-gray"/>
```

Current behaviour only allows this:
```js
// base className
<div className="dib pa3 color-white"/>

// selected
<div className="dib pa3 color-white color-blue"/>

// disabled
<div className="dib pa3 color-white color-gray"/>
```

**In other words:**
The current setup of react-tabs assumes a BEM style of classes where you have a base class and augment it depending on the state. This is fine because you can take care of CSS precedence with three BEM classes. With atomic/functional classes (and I am sure other use cases), this is not necessarily the case. _This PR enables disabling automatic BEM behaviour._

**Design notes:**
I called it `noBEM` to show that the BEM behaviour is not specified. It is up to the caller to make sure that the base class is repeated in the selected and disabled classes, if needed. It could be called `noAutoBEM` `disableAutoBEM`, or anything you deem appropriate.

I also chose this prop to be opt-in, so as not to break current behaviour. Apps probably rely on the automatic BEM style already, and I don't want to force a semver major bump.

There is also an _alternative design_, where instead of `noBEM`, we introduce an optional `idleClassName`, which is true on `!disabled && !selected`. This may be more general. I can do that if you think it fits your plans for the library better.

I hope I'm not missing anything, but I could not find a `Contributing.md` or equivalent. If you have any changes to suggest (tests, naming things), I'd be happy to work on them :)

### CODE REVIEW

**Comment by joepvl** [comment]
I understand what you'd like to achieve, but I don't think it's a common enough use case to warrant first-class support for it. To be honest, to me it seems like an anti-pattern CSS-wise, but I can see how that's debatable. If anything, the `idleClassName` you're proposing seems like a much better solution, but again I don't think we should add it to the package.

However, all is not lost! You can solve your problem by writing a thin wrapper around `Tab` and `TabPanel` that adds an `idleClassName` prop, and using that instead inside `Tabs` and `TabList`. I've created an example for you on codesandbox: https://codesandbox.io/s/pjl902plr7.

TL;DR:
```jsx
class IdleTab extends Component {
  render() {
    const { className, idleClassName, ...props } = this.props;
    const cn = cx(className, { [idleClassName]: !props.disabled && !props.selected })
    return <Tab {...props} className={cn} />
  }
}
IdleTab.tabsRole = 'Tab';

class IdleTabPanel extends Component {
  render() {
    const { className, idleClassName, ...props } = this.props;
    const cn = cx(className, { [idleClassName]: !props.selected })
    return <TabPanel {...props} className={cn} />
  }
}
IdleTabPanel.tabsRole = 'TabPanel';

const App = () => (
  <Tabs>
    <TabList>
      <IdleTab idleClassName='foo'>Tab 1</IdleTab>
      <IdleTab idleClassName='foo' disabled>Tab 2</IdleTab>
      <IdleTab idleClassName='foo'>Tab 3</IdleTab>
    </TabList>
    <IdleTabPanel idleClassName='bar'>Panel 1</IdleTabPanel>
    <IdleTabPanel idleClassName='bar'>Panel 2</IdleTabPanel>
    <IdleTabPanel idleClassName='bar'>Panel 3</IdleTabPanel>
  </Tabs>
);
```

...and `idleClassName` will only be added to the `className` if the tab is neither selected nor disabled.

**Comment by fpapado** [comment]
Having slept on it a bit, I agree that `idleClassName` is the real wish here, to separate "core class" and "this is what happens when the state is idle | selected | disabled". Currently, you are encouraged to keep idle and core together (which seems reasonable tbh), so something like `Tab Tab--idle` as base leads to `Tab Tab--idle Tab--disabled`, which still relies on precedence. I thus think that adding `idleClassName` is consistent with the other state className props, but I understand if it doesn't fit the vision atm.

Regardless, thank you for the example! The wrapper helps our use case very much 🎉 

(Feel free to close this as you see fit)

**Comment by joepvl** [comment]
Alright then! Well, since I was just added as a maintainer on this repo, my first action in that role will be to close this PR. 🎉

I do want to make it abundantly clear that we really appreciate the contributions (both code and comments) that you made here, and all the work and thought you put into them. Before this I had never even considered your angle, and I do agree it's worth keeping in mind.

---

P.S. My personal 2c, as a reply to your last comment: I personally use the SUIT CSS approach with react-tabs, with classes like this:

- `Tab` for the base/idle state
- `Tab is-disabled` for disabled state
- `Tab is-selected` for selected state

Since tabs are never in a "none of these things" state, we don't need to define "idle" separately here — I don't think separating the two adds value. Since state-specific (`is-*`) classes always have higher specificity anyways (they're always defined in CSS like `.Tab.is-selected`, never by themselves), they can override what they need to without extra effort. 
I can see where I could theoretically run into an issue if I was using a utility class (i.e. something like `u-alignLeft`) on `Tab`. Honestly I don't have a better answer to that than that it just seems pragmatic to me to only use them when they serve the use case, which IMO they wouldn't here. 😉
